source/dem/particle_particle_contact_force.cc

```diff
@@ -17,25 +17,984 @@
+                    {
+                      calculate_hertz_mindlin_limit_force_contact(
+                        contact_info,
+                        this->normal_relative_velocity_value,
+                        this->normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        this->normal_force,
+                        this->tangential_force,
+                        this->particle_one_tangential_torque,
+                        this->particle_two_tangential_torque,
+                        this->rolling_resistance_torque);
+                    }
+
+                  if constexpr (contact_model ==
+                                Parameters::Lagrangian::
+                                  ParticleParticleContactForceModel::
+                                    hertz_mindlin_limit_overlap)
+                    {
+                      calculate_hertz_mindlin_limit_overlap_contact(
+                        contact_info,
+                        this->normal_relative_velocity_value,
+                        this->normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        this->normal_force,
+                        this->tangential_force,
+                        this->particle_one_tangential_torque,
+                        this->particle_two_tangential_torque,
+                        this->rolling_resistance_torque);
+                    }
+
+
+                  types::particle_index particle_two_id =
+                    particle_two->get_local_index();
+
+                  Tensor<1, 3> &particle_two_torque = torque[particle_two_id];
+                  Tensor<1, 3> &particle_two_force  = force[particle_two_id];
+
+                  // Apply the calculated forces and torques on the particle
+                  // pair
+                  this->apply_force_and_torque_on_local_particles(
+                    this->normal_force,
+                    this->tangential_force,
+                    this->particle_one_tangential_torque,
+                    this->particle_two_tangential_torque,
+                    this->rolling_resistance_torque,
+                    particle_one_torque,
+                    particle_two_torque,
+                    particle_one_force,
+                    particle_two_force);
+                }
+
+              else
+                {
+                  // if the adjacent pair is not in contact anymore, only the
+                  // tangential overlap is set to zero
+                  for (int d = 0; d < dim; ++d)
+                    {
+                      contact_info.tangential_overlap[d] = 0.0;
+                    }
+                }
+            }
+        }
+    }
+
+  // Doing the same calculations for local-ghost particle pairs
```

Comment:
I agree with you there, but I would rather do that in a next PR. I prefer to go step by step since I know my time is limited and I won't be able to dedicate hours to this before next week.

Reply:
No problems :)